interfaces/apparmor/template_vars.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package apparmor
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/ubuntu-core/snappy/interfaces"
+	"github.com/ubuntu-core/snappy/interfaces/dbus"
+	"github.com/ubuntu-core/snappy/snap"
+)
+
+// legacyVariablees returns text defining some apparmor variables that work
+// with legacy apparmor templates.
+//
+// The variables are expanded by apparmor parser. They are (currently):
+//  - APP_APPNAME
+//  - APP_ID_DBUS
+//  - APP_PKGNAME_DBUS
+//  - APP_PKGNAME
+//  - APP_VERSION
+//  - INSTALL_DIR
+// They can be changed but this has to match changes in template.go.
+//
+// In addition, the set of variables listed here interacts with old-security
+// interface since there the base template is provided by a particular 3rd
+// party snap, not by snappy.
+func legacyVariables(appInfo *snap.AppInfo) string {
+	var buf bytes.Buffer
+	fmt.Fprintf(&buf, "@{APP_APPNAME}=\"%s\"\n", appInfo.Name)
+	// TODO: replace with app.SecurityTag()
+	fmt.Fprintf(&buf, "@{APP_ID_DBUS}=\"%s\"\n",
+		dbus.SafePath(fmt.Sprintf("%s.%s_%s_%s",
+			appInfo.Snap.Name, appInfo.Snap.Developer, appInfo.Name, appInfo.Snap.Version)))
+	// XXX: How is this different from APP_ID_DBUS?
+	fmt.Fprintf(&buf, "@{APP_PKGNAME_DBUS}=\"%s\"\n",
+		dbus.SafePath(fmt.Sprintf("%s.%s", appInfo.Snap.Name, appInfo.Snap.Developer)))
+	// TODO: stop using .Developer, investigate how this is used.
+	fmt.Fprintf(&buf, "@{APP_PKGNAME}=\"%s.%s\"\n", appInfo.Snap.Name, appInfo.Snap.Developer)
+	// TODO: switch to .Revision
+	fmt.Fprintf(&buf, "@{APP_VERSION}=\"%s\"\n", appInfo.Snap.Version)
+	fmt.Fprintf(&buf, "@{INSTALL_DIR}=\"{/snaps,/gadget}\"\n")
+	return buf.String()
+}
+
+// modenVariables returns text defining some apparmor variables that
+// work with non-legacy apparmor templates.
+//
+// XXX: Straw-man: can we just expose the following apparmor variables...
+//
+// @{APP_NAME}=app.Name
+// @{APP_SECURITY_TAG}=app.SecurityTag()
+// @{SNAP_NAME}=app.SnapName
+//
+// ...have everything work correctly?
+func modernVariables(appInfo *snap.AppInfo) string {
+	var buf bytes.Buffer
+	fmt.Fprintf(&buf, "@{APP_NAME}=\"%s\"\n", appInfo.Name)
+	fmt.Fprintf(&buf, "@{APP_SECURITY_TAG}=\"%s\"\n", interfaces.SecurityTag(appInfo))
+	fmt.Fprintf(&buf, "@{SNAP_NAME}=\"%s\"\n", appInfo.Snap.Name)
```

Comment:
Introducing these now confuses policy since nothing in policy references these variables. I don't even know why we are exposes APP_SECURITY_TAG-- what is that? I suspect it is the profile name, but that is unneeded since apparmor already has a special @{profile_name} variable that the parser automatically creates.

I would prefer all but INSTALL_DIR be removed until this is properly choreographed with template.go and all the bultin/ policy.